docs: update stake requirements (#947)
oasisprotocol/docs: docs/node/run-your-node/prerequisites/stake-requirements.md

# Stake Requirements

This page provides an overview of the stake requirements to become a validator 
on the consensus layer and to run a ParaTime compute node of the Oasis Network.

## Consensus Layer

### Mainnet

To become a validator on the consensus layer, you will need to have enough ROSE
tokens staked in your escrow account.

Currently, you are going to need:

* 100 ROSE staked for your entity's registration, since this is the
[staking threshold].
* 100 ROSE staked for your validator node's registration, since this is the
[staking threshold].
* If you want to be part of the active validator set, you will need enough
ROSE staked --  this means you must be one of the top 120 entities
by stake. You can check the current top 120 entities on blockchain explorers

> like [Oasis Scan].

Reviewer: FYI the Testnet's maximum validator set size is currently `110`.

# Stake Requirements

This page provides an overview of the stake requirements to become a validator
on the consensus layer and to run a ParaTime compute node of the Oasis Network.

## Consensus Layer

> To become a validator on the consensus layer, you will need to have enough ROSE

Reviewer: ```suggestion
To become a validator on the consensus layer, you will need to have enough ROSE
for **registering your entity**, **registering your validator node**, and be among
the top entities by stake to be included in the **validator set**.
```

tokens staked in your escrow account.

Currently, you are going to need:

|                             | Mainnet  | Testnet  |
| --------------------------- | -------- | -------- |
| Entity Registration         | 100 ROSE | 100 TEST |
| Validator Node Registration | 100 ROSE | 100 TEST |
| Validator Set Size          | 120      | 110      |

> If you want to be part of the active validator set, you will need enough
> ROSE/TEST staked --  this means you must be one of the top 120 entities (respectively top 110 on the testnet)
> by stake. You can check the current top validators on blockchain explorers
> like [Oasis Scan].

Reviewer: ```suggestion
You can check the current top validators on the blockchain explorers like the [Oasis Scan].

```

> For more information on obtaining information about your entity's account, see
> the [Account Get Info] doc.
> 
> Please note that staking thresholds may change in the future. Refer to the
> [native token information] to see the current values used by the network.

Reviewer: ```suggestion
::: tip

You can use the Oasis CLI to check your account and the network parameters. Use
the [`oasis account show`] command to check your entity's account and
[`oasis network show native-token`] to see the current values used by the network.

:::
```

> For the testnet, you can receive a limited number of TEST tokens by using our [Oasis Network Testnet Faucet][faucet-testnet].

Reviewer: ```suggestion
For the Testnet, you can receive a limited number of TEST tokens by using our [Oasis Network Testnet Faucet][faucet-testnet].
```

:::info

To determine if you are eligible to receive a delegation from the Oasis Protocol
Foundation, see the [Delegation Policy] document.

:::

:::info

The size of the consensus committee (i.e. the validator set) is configured by
the [**max_validators** consensus parameter][**max_validators** consensus parameter].

:::

For the Testnet, you can receive a limited number of TEST tokens by using our [Oasis Network Testnet Faucet][faucet-testnet].

> For more tokens please contact us on our official [Discord in the #testnet channel][discord].

Reviewer: I would move this above the admonition boxes.

Author: Yes, I was not sure. My thought was the boxes are more important and should be put first, but like this probably no one ever will read the faucet paragraph.

## ParaTime

To run a ParaTime compute node, the minimum you need to stake currently is:

|            | Mainnet     | Testnet  |
| ---------- | ----------- | -------- |
| Sapphire   | 50,000 ROSE | 100 TEST |
| Emerald    | 50,000 ROSE | 100 TEST |
| Cipher     | 100 ROSE    | 100 TEST |

> You can receive a limited number of TEST tokens by using our [Oasis Network Testnet Faucet][faucet-testnet].
> For more tokens please contact us on our official [Discord in the #testnet channel][discord].

Reviewer: You only need TEST tokens on consensus to run a Sapphire node. So I would remove mentioning the faucet here.

```suggestion
```

> [Oasis Scan]: https://www.oasisscan.com/validators
> [Account Get Info]: ../../../general/manage-tokens/cli/account.md#show
> [native token information]: ../../../general/manage-tokens/cli/network.md#show-native-token

Reviewer: Move these at the end of the `## Consensus` section since that's where they are used.

```suggestion
[Oasis Scan]: https://www.oasisscan.com/validators
[`oasis account show`]: ../../../general/manage-tokens/cli/account.md#show
[`oasis network show native-token`]: ../../../general/manage-tokens/cli/network.md#show-native-token
```

:::

### Testnet

For the Testnet you are going to need TEST tokens staked in your escrow account.

Currently, you are going to need:

* 100 TEST staked for your entity's registration since that is the

> [staking thresholds].

Reviewer: ```suggestion
[staking threshold].
```

* 100 TEST staked for your validator node's registration since that is the

> [staking thresholds].

Reviewer: ```suggestion
[staking threshold].
```
